data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/RecordDispatcherImpl.java

```diff
@@ -295,7 +296,8 @@ private ConsumerRecordContext errorTransform(final ConsumerRecordContext recordC
       if (data.length() > KN_ERROR_DATA_MAX_BYTES) {
         data = data.substring(0, KN_ERROR_DATA_MAX_BYTES);
       }
-      extensions.put(KN_ERROR_DATA_EXT_NAME, data);
+
+      extensions.put(KN_ERROR_DATA_EXT_NAME, Base64.getEncoder().encodeToString(data.getBytes()));
```

Comment:
One problem with transforming the response body to string and then base64'ing it is that it might produce unserializable bytes on the other side.

For example, when the body is a protobuf message, converting bytes to string and then base64'ing it will cause the message to be unserializable, is there any way to base64 the response body as bytes?

Reply:
To clarify:

we do:
```java
data = response.bodyAsString();
Base64.getEncoder().encodeToString(data.getBytes())
```

Reply:
Good point, I think there is some vertx byte buffer/array